File: 1erdepartamental/c++/21.quick_indirecto/main.cpp
#include <iostream>
#include "operations/OrdenadorIndirectoQuick.h"

int main() {
    std::vector<int> numeros = {64, 34, 25, 12, 22, 11, 90};
    
    OrdenadorIndirectoQuick<int> ordenador(numeros);
    
    ordenador.mostrarOriginal();
    ordenador.ordenar();
    ordenador.mostrarOrdenado();
    ordenador.mostrarOriginal();
    
    return 0;
}

File: 1erdepartamental/python/21.quick_indirecto/main.py
"""
Programa: Quick Indirecto
Demuestra quick sort sin modificar el arreglo original
"""
# Importa la clase 
from core.ordenador import OrdenadorIndirectoQuick
#Punto de entrada 
if __name__ == "__main__":
    numeros = [64, 14, 25, 43, 22, 11, 90]

    ordenador = OrdenadorIndirectoQuick(numeros)

    print(f"Original: {ordenador.mostrar_original()}")
    ordenador.ordenar()
    print(f"Ordenado (mediante índices): {ordenador.mostrar_ordenado()}")
    print(f"Original (sin cambios): {ordenador.mostrar_original()}")
